trin-core/src/utils/binary_string.rs

```diff
@@ -0,0 +1,66 @@
+use rand::Rng;
+use std::num::ParseIntError;
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum BinaryStringError {
+    #[error("Unable to parse binary string!")]
+    ParseError(#[from] ParseIntError),
+}
+
+/// Generate random 256-bit binary string with leading zeroes
+pub fn generate_random(leading_zeroes: u8) -> Result<String, BinaryStringError> {
```

Comment:
When would this ever return a `Result::Err`?

Reply:
Ah, this is a leftover from some previous versions of the method. It is fixed now.